src/components/Layout/NavigationLayout/GenreNavigation/GenreNavigation.tsx

```diff
@@ -1,11 +1,28 @@
-import { ALL_FILTERS, GENRE_FILTER } from '../../../../constants/constants';
+import { useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+
+import { ALL_FILTERS, FILTER_BY, GENRE_FILTER } from '../../../../constants/constants';
 import { moviesActions } from '../../../../redux/moviesSlice';
-import { useAppDispatch, useAppSelector } from '../../../../redux/store';
+import { useAppDispatch } from '../../../../redux/store';
+import { useQuery } from '../../../common/hooks/useQuery';
 
 import './GenreNavigation.scss';
 
 const GenreNavigation = (): JSX.Element => {
-  const filters = useAppSelector((state) => state.movies.filters);
+  const query = useQuery();
+  const genreFilterValue: string = query.get(FILTER_BY);
+  const getFilters = (): string[] => {
+    let appliedFilters = [];
+    if (genreFilterValue?.toLowerCase().split(',')) {
+      appliedFilters = genreFilterValue?.toLowerCase().split(',');
+    }
+    return appliedFilters;
+  };
+
+  const [filters, setFilters] = useState(getFilters());
+  const location = useLocation();
+  const history = useHistory();
+  const queryParams = new URLSearchParams(location.search);
```

Comment:
There is a `query` already that you can work with

Reply:
sure